index/language-settings.go

```diff
@@ -0,0 +1,97 @@
+package index
+
+import (
+	"fmt"
+	"strings"
+)
+
+type LanguageConfig struct {
+	Name     string
+	Language string
+	Settings OpensearchSettings
+}
+
+type OpensearchSettings struct {
+	Settings struct {
+		Analysis struct {
+			Analyzer struct {
+				Default struct {
+					Type string `json:"type"`
+				} `json:"default"`
+			} `json:"analyzer"`
+		} `json:"analysis"`
+	} `json:"settings"`
+}
+
+func GetLanguageConfig(code string) (LanguageConfig, error) {
+	parts := strings.Split(code, "-")
+	if len(parts) != 2 {
```

Comment:
This introduces a new constraint on document languages that we haven't had before: we require all language codes on documents to be "fully qualified" with region. I'm not **completely** opposed, but I think that might be a mistake to force that kind of specificity. Most of the time the regional distinction won't be made, and content will be written in a mix with forced mislabeling.

My example left that more open:

```go
	analyzer, exactMatch := analyzers[code]
	if exactMatch {
		return analyzer
	}

	language, _, _ := strings.Cut(code, "-")

	for c := range analyzers {
		if c == language {
			return analyzers[c]
		}
	}
```

Either way, that would need to be enforced by the repository as well, otherwise we'll just end up with un-indexable documents.

Reply:
Turns out that was pretty easy to achieve. The language matcher now accepts codes without a region (`sv`, `en` etc.)